### PR TITLE
Docker build

### PR DESCRIPTION
**Pull request by cratliff**
snapcraft was installed from apt or the snap store then placed in a docker file.  Modified dockerfile to use the local source as opposed to remote sources.

---
This change is [Reviewable](https://reviewable.io/reviews/badgertechnologies/snapcraft/1)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,13 +5,32 @@ RUN sed -i 's/ universe/ universe multiverse/' /etc/apt/sources.list
...
-  snapcraft \
-  && \
   apt-get autoclean --yes && \
   apt-get clean --yes
 
+# Install snapcraft dependencies
+RUN apt-get install git sudo libslang2-dev python-pip python3-pip gcc g++ make python3-dev python3-venv libffi-dev \
+  libsodium-dev libapt-pkg-dev libarchive13 squashfs-tools patchelf --yes && \
+  apt-get autoclean --yes && \
+  apt-get clean --yes
```
Comment (hensleyl) [inline Dockerfile:15]: You might get a smaller image if you combine this RUN with the one above and add `rm -rf /var/lib/apt/lists/*` at the end.

File: Dockerfile [hunk 2]
```diff
@@ -5,13 +5,32 @@ RUN sed -i 's/ universe/ universe multiverse/' /etc/apt/sources.list
...
+. /venv/snapcraft/bin/activate && \
+pip --no-cache-dir install wheel && \
+pip --no-cache-dir install -r /snapcraft/requirements.txt
+
+RUN echo """#!/bin/bash\n\
+    \n\
+    . /venv/snapcraft/bin/activate\n\
+    python3 /snapcraft/bin/snapcraft \$@\n\
+    deactivate\
+    """ > /bin/snapcraft && chmod +x /bin/snapcraft
```
Comment (hensleyl) [inline Dockerfile]: This could just be a file in the repo that you copy into place

**Review by hensleyl — APPROVED** [review]
Looks good.  Feel free to ignore my comments

File: Dockerfile (hunk 2 above)
Comment (cratliff) [reply Dockerfile]: :+1:

File: Dockerfile (hunk 1 above)
Comment (cratliff) [reply Dockerfile:15]: Removing those files seems to break things since snapcraft makes use of apt when it runs so I'm going to skip it.

**Pull request merged**